xapian-applications/omega/docs/omegascript.rst

```diff
@@ -222,10 +222,10 @@ $fmt
 $freq{term}
 	frequency of a term
 
-$hash{TEXT[,HASH}
+$hash{TEXT[,HASH]}
     unique ID for ``TEXT`` string using the hashing algorithm specified by
-    ``HASH`` which defaults to MD5 hashing algorithm. Currently, this command
-    only supports MD5 which yields a 128-bit hash sequence of 32 hexadecimal
+    ``HASH`` which defaults to MD5. Currently, this command only supports
+    MD5 which yields a 128-bit hash sequence serialised as 32 hexadecimal
     characters.
```

Comment:
> If we start talking about security, we probably shouldn't use md5 at all.

I think the only reason we are at all is in the interests of expediency: there's already code to calculate MD5 hashes in the tree so it's much simpler to hook that up.

> there's something a little weird about having an option with only one valid choice, which nonetheless has to be provided

True, but I'd expect we'll add more options in the near future.  I'm quite keen to find a sensible way to switch to something else for duplicate removal - if a system hides duplicates based on their hash, currently someone who can add documents to that system can use hash collisions to hide results.  And once we have another hash function available we can hook that up in omega for little extra work.

> Just to be sure, should we drop the default behaviour then?

I think we should.  The main reasons I see for having a single `$hash` are (a) to control the proliferation of omegascript functions a bit (b) it means that choice of hash can be easily parameterised (if instead we had separate functions for `$md5`, `$sha256`, `$blake2b`, etc then you'd have to use `$eval` to achieve that with the same generality, which is problematic as it is essentially executing the configuration file setting as code).